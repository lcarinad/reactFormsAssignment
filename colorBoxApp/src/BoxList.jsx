//renders Box components and New Box Form
import React, { useState } from "react";
import Box from "./Box";
import NewColorBoxForm from "./NewColorBoxForm";
import { v4 as uuid } from "uuid";

const BoxList = () => {
  const [boxes, setBoxes] = useState([]);

  const addBox = (newBoxFormData) => {
    setBoxes((boxes) => [...boxes, { ...newBoxFormData, id: uuid() }]);
  };

  const removeBox = (id) => {
    setBoxes((boxes) => boxes.filter((box) => box.id != id));
  };
  return (
    <div>
      <h1>Box List</h1>
      <NewColorBoxForm addBox={addBox} />
      <div>
        {boxes.map(({ id, backgroundColor, width, height }) => (
          <Box
            handleRemove={removeBox}
            id={id}
            backgroundColor={backgroundColor}
            width={width}
            height={height}
            key={id}
          />
        ))}
      </div>
    </div>
  );
};

export default BoxList;
