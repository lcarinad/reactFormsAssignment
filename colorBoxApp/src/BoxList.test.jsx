import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

function addBox(boxList, height = "2", width = "2", color = "peachpuff") {
  const heightInput = boxList.getByLabelText("Height");
  const widthInput = boxList.getByLabelText("Width");
  const backgroundInput = boxList.getByLabelText("Background ColorðŸŒˆ");
  fireEvent.change(backgroundInput, { target: { value: color } });
  fireEvent.change(widthInput, { target: { value: width } });
  fireEvent.change(heightInput, { target: { value: height } });
  const button = boxList.getByText("Add a new box!");
  fireEvent.click(button);
}

it("renders without crashing", function () {
  render(<BoxList />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

describe("adding box", function () {
  it("should add new box", function () {
    const BoxList = render(<BoxList />);
    //no boxes
    expect(BoxList.getByClass("Box")).not.toBeInTheDocument();
    const input = getByLabelText("Height");
    const btn = queryByText("Create Color Box");
    expect(queryByClassName("Box")).not.toBeInTheDocument();

    addBox(BoxList);

    //expect to see box
    const removeButton = BoxList.queryByClassName("remove");
    expect(removeButton).toBeInTheDocument();
  });
});
