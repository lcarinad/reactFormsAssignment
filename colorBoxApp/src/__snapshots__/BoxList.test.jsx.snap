// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Box List
    </h1>
    <form
      class="AddBoxForm"
    >
      <label
        for="backgroundColor"
      >
        Background ColorðŸŒˆ: 
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        placeholder="background color"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Width: 
      </label>
      <input
        id="width"
        name="width"
        placeholder="width"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        Height: 
      </label>
      <input
        id="height"
        name="height"
        placeholder="height"
        type="text"
        value=""
      />
      <button>
        Create Color Box
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
